src/blockchain/pop/vbk_block_tree.cpp

```diff
@@ -451,6 +437,25 @@ bool VbkBlockTree::isStronglyEquivalent(const VTB& vtb1, const VTB& vtb2) {
          are_on_the_same_chain;
 }
 
+bool VbkBlockTree::isWeaklyEquivalent(const VTB& vtb1, const VTB& vtb2) {
+  bool is_on_the_same_btc_chain = btc().areOnSameChain(
+      vtb1.transaction.blockOfProof, vtb2.transaction.blockOfProof);
+
+  bool is_on_the_same_vbk_chain = areOnSameChain(
+      vtb1.transaction.publishedBlock, vtb2.transaction.publishedBlock);
+
+  bool is_the_same_keystone_period =
+      abs(vtb1.transaction.publishedBlock.height -
+          vtb2.transaction.publishedBlock.height) <=
+      (int32_t)getParams().getKeystoneInterval();
```

Comment:
this is wrong.
Correct would be to get keystone height for each published block, and then == compare.

Example: keystone interval is 20 (in VBK), 19 and 21 are on different keystones, but this comparison would return true